src/structures/BaseCommandInteraction.js

```diff
@@ -74,6 +75,64 @@ class BaseCommandInteraction extends Interaction {
     return this.guild?.commands.cache.get(id) ?? this.client.application.commands.cache.get(id) ?? null;
   }
 
+  /**
+   * Represents the resolved data of a received command interaction.
+   * @typedef {Object} CommandInteractionResolvedData
+   * @property {Collection<string, User>} users The resolved users
+   * @property {Collection<string, GuildMember|APIGuildMember>} members The resolved guild members
+   * @property {Collection<string, Role|APIRole>} roles The resolved messages
+   * @property {Collection<string, Channel|APIChannel>} channels The resolved channels
+   * @property {Collection<string, Message|APIMessage>} messages The resolved messages
+   */
+
+  /**
+   * Transforms the resolved received from the API.
+   * @param {APIInteractionDataResolved} resolved The received resolved objects
+   * @returns {CommandInteractionResolvedData}
+   * @private
+   */
+  transformResolved({ members, users, channels, roles, messages }) {
+    const result = {};
+
+    if (members) {
+      result.members = new Collection();
+      for (const [id, member] of Object.entries(members)) {
+        const user = users[id];
+        result.members.set(id, this.guild?.members._add({ user, ...member }) ?? member);
```

Comment:
i think there should be a check to see if `guild` is not-null before adding to the collection, seems a bit silly to add `undefined` to the collection (even if its rare)